### ISSUE TITLE
[1] Feed: Create a Filter Query that sets a foundation to extend the code for future filters

### ISSUE THREAD

**Issue by Ridhwana** [forem/forem#20160]
## Is this feature related to a problem? Please describe.

As a community member, I want my feed to be refined as possible which includes being able to filter by aspects that relate to me like:
- showing only content from users and organizations that I'm following
- showing posts in the languages that I understand
- hiding "hidden" tags across all my feeds.

## Describe the solution you’d like

In order to do this, we need to update the architecture to be as extensible and flexible as possible. The way that we'd like to accomplish it is to think about the feed as a collection of items that we filter by certain parameters in order to get to a final subset of data. We've taken an attempt at creating a [POC](https://github.com/forem/forem/pull/20144/files) for this pattern to test its feasibility.

Similarly to the way that we filter billboards, we've implemented a [Articles::Feeds::FilterQuery in this POC](https://github.com/forem/forem/pull/20144/files#diff-d3bed6f8cfbde3f4d9a04a6bb2b1556d69bfc329a92cafa4baf0e2d17c56a843). In this query we're able to filter by properties properties and re-use the class in other places. The POC linked covers the Latest/Timeframe feed and some other filters like hidden tags and following tags( which should have been following users/organizations).

It may be possible to migrate Articles::Feeds::Basic to use [the Articles::Feeds::FilterQuery](https://github.com/forem/forem/pull/20144/files#diff-d3bed6f8cfbde3f4d9a04a6bb2b1556d69bfc329a92cafa4baf0e2d17c56a843) and I would advise trying that so that we can handle Relevant/Latest/Top in one place. However, I think it might be harder to migrate VariantQuery and ForemLargeExperimental (not really relevant here) to migrate to using this query. Hence we should keep those as is.  

TLDR; Articles::Feeds::FilterQuery will be responsible for filtering the feed in order to produce the subset of relevant data.

**_NOTE: We will not be implementing the "Following" Feed in this issue but rather setting up the foundation of the Query so that adding the "Following" filter becomes a more achievable task._** 
 
## Definition of Done

- [ ] Create an `Articles::Feeds::FilterQuery` that filters the feed to display the relevant result. In this class we want to implement the basic filtering principles like filtering by latest, timeframe, tags, following etc. As mentioned, I've taken a stab of what this can look like [here](https://github.com/forem/forem/pull/20144/files#diff-d3bed6f8cfbde3f4d9a04a6bb2b1556d69bfc329a92cafa4baf0e2d17c56a843). 
- [ ] Migrate all uses of `Articles::Feeds::Latest` and `Articles::Feeds::Timeframe` to use `Articles::Feeds::FilterQuery`
- [ ] Attempt to migrate `Articles::Feeds::Basic` to use ` Articles::Feeds::FilterQuery`.
- [ ] Migrate `Articles::Feeds::Tags` to use `Articles::Feeds::FilterQuery`
- [ ] Migrate the signed out version of the feed to use `Articles::Feeds::FilterQuery` for `Articles::Feeds::Latest` and `Articles::Feeds::Timeframe`.
- [ ] Update the routes accordingly to cater for the feedType. `/` and `/explore` will need to point to Explore < Relevant

Some small cleanup that will help make the feed more coherent:
- [ ] Rename `Articles::Feeds::Basic` to `Articles::Feeds::RecommendedBasic`
- [ ] Rename `Articles::Feeds::VariantQuery` to `Articles::Feeds::RecommendedAdvanced`
- [ ] If you see something not aptly names, go ahead and rename it.

## Additional Context

We'll need to ensure that this is fully tested both through tests in the codebase as well putting it on some staging env and Qa'ing manually before rolling out.
